inc/question.class.php

```diff
@@ -866,25 +866,33 @@ public static function uninstall()
    }
 
    /**
-    * Clone a question
-    * @param  array  $input with these keys
-    *                       - id the id of the question to clone
-    * @return integer the question id of the new clone
+    * Duplicate a question
+    *
+    * @return boolean
     */
-   public function cloneItem($input = []) {
-      global $DB;
-
-      if ($DB->isSlave()) {
+   public function duplicate() {
+      $oldQuestionId       = $this->getID();
+      $newQuestion         = new static();
+      $question_condition  = new PluginFormcreatorQuestion_Condition();
+
+      $row = $this->fields;
+      unset($row['id'],
+            $row['uuid']);
+      if (!$newQuestion->add($row)) {
          return false;
       }
 
-      if (!$this->getFromDB($input[static::getIndexName()])) {
-         return false;
+      // Form questions conditions
+      $rows = $question_condition->find("`plugin_formcreator_questions_id` IN  ('$oldQuestionId')");
+      foreach($rows as $conditions_id => $row) {
+         unset($row['id'],
+               $row['uuid']);
+         $row['plugin_formcreator_questions_id'] = $newQuestion->getID();
```

Comment:
Here is how I designed the show_field update process

For form duplication (done in #389) I update all show_fields because the whole form is duplicated. This ensure show_field is not cross-form.

For section duplication (in this PR) I update show_field only if the referenced question ID is duplicated. If not I keep the original question ID.

For single question duplication (in this PR), show_field always references an other question if the same form, then I think it shold not be changed (by the way show_field always different of the being duplicated question ID)